### PR TITLE
ocserv: Correctly get runtime LAN ifname and addresses

### PR DESCRIPTION
**Pull request by billsq**
Maintainer: @billsq
Compile tested: (mvebu, WRT1900ACS, LEDE 17.01.1)
Run tested: (mvebu, WRT1900ACS, LEDE 17.01.1)

Description:
1. Correctly get LAN runtime ifname and addresses using network functions.
2. Do not store ip settings in config file as they may change next time.

Signed-off-by: Qian Sheng <billsq@billsq.me>

### CODE REVIEW

**Comment by hnyman** [comment]
You should also increase PKG_RELEASE in Makefile.

cc to maintainer: @nmav

**Comment by nmav** [comment]
I'm fine with it. @hnyman feel free to merge once the PKG_RELEASE is bumped (I'll be unavailable for few days)

**Comment by nmav** [comment]
Thank you.

**Comment by hnyman** [comment]
@billsq 
You need to include the "signed-off-by" line in the commit message itself, as the commit message gets stored to git history unlike this dicussion. And the commit title should start with "ocserv: ". Please amend the commit.  

At the same time you can squash the two commits together, so that there is just one commit.

Not sure if you need git advice how to do it, but advice can be found in https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md#advice-on-pull-requests

Short version: you have the commits in a branch, so it is pretty easy. 
* checkout the branch
* then do "git rebase -i HEAD~2" and mark the second commit squashed, close rebase editor, and then you can edit the combined commit message to contain the signed-off-by line .
* The pull request can then be refreshed by force-pushing the branch. Just do "git push -f".

**Comment by billsq** [comment]
Thanks @hnyman, I've squashed the commits but the pull request doesn't seem to be updated. 
Anyway your advices are very helpful. Coming from perforce world all git stuff is new to me.

**Comment by hnyman** [comment]
Oh yes, my bad. 
nmav already merged this (before I wrote my comments), so there is no need to do anything anymore. Just delete the branch.

File: net/ocserv/files/ocserv.init [hunk 1]
```diff
@@ -33,26 +35,27 @@ setup_config() {
...
+			# Append ipv6 prefix
+			test -n "$ip6addr" && ip6addr="$ip6addr/96"
 		fi
 
 		ping_leases=1
-		test -n "$ipaddr" && sysctl -w "net.ipv4.conf.$(uci get network.lan.ifname).proxy_arp"=1 >/dev/null
-		test -n "$ip6addr" && sysctl -w "net.ipv6.conf.$(uci get network.lan.ifname).proxy_ndp"=1 >/dev/null
+		local ifname
+		network_get_device ifname lan
+		if test -n "ifname";then
```
Comment (jow-) [inline net/ocserv/files/ocserv.init]: This looks like a typo and should likely be `test -n "$ifname"`.

You can also shorten the code into just

    if network_get_device ifname lan; then
        ...
    fi

Comment (billsq) [inline net/ocserv/files/ocserv.init]: Yes this is indeed a typo... Didn't catch this since ifname always has a value.
